Library/OcMachoLib/Header.c

```diff
@@ -97,35 +97,54 @@ MachoGetVmSize64 (
 }
 
 /**
-  Moves file pointer and size to point to x86_64 slice in case
-  FAT Mach-O is used.
-
-  @param[in,out] FileData  Pointer to pointer of the file's data.
-  @param[in,out] FileSize  Pointer to file size of FileData.
+  Returns offset and size of specified slice in case
+  FAT Mach-O is used. If no FAT is detected, FatOffset and
+  FatSize are set to 0 and FullSize respectively.
+
+  @param[in]  Buffer      Pointer to the buffer data.
+  @param[in]  BufferSize  Size of Buffer.
+  @param[in]  FullSize    Full size of Buffer, used to validate sizes
+                          within FAT structure.
+  @param[in]  CpuType     Desired CPU slice to use.
+  @param[out] FatOffset   Pointer to offset of FAT slice.
+  @param[out] FatSize     Pointer to size of FAT slice.
 
   @return FALSE is not valid FAT image.
 **/
-STATIC
 BOOLEAN
-MachoFilterFatArchitecture64 (
-  IN OUT UINT8         **FileData,
-  IN OUT UINT32        *FileSize
+MachoGetFatArchitectureOffset (
```

Comment:
Rename to `FatGetArchitectureOffset`, as it is not necessarily Mach-O. Also drop the docs, as they are already present in .h file.

Make it return TRUE:
- when the file is a FAT and the architecture is present
- when the file is not a FAT

Otherwise FALSE. Same for the functions below.